src/core/shapes/Stomatocyte.cpp

```diff
@@ -33,41 +33,28 @@ void Stomatocyte::calculate_dist(const Utils::Vector3d &pos, double &dist,
 
   using Utils::sqr;
 
-  // Parameters
-
-  int number;
-
-  double mu, T0, T1, T1p, T2, T3, T3sqrt, T3p, T4, a, b, c, d, e, rad0, rad1,
-      rad2, rad3, dst0, dst1, dst2, dst3, t0, t1, t2, t3, t4, ttota, distance,
-      mindist, time0, time1, xd, yd, zd, normal_x_3D, normal_y_3D, normal_z_3D;
-
-  Utils::Vector3d closest_pos({-1.0, -1.0, -1.0});
-
   // Set the three dimensions of the stomatocyte
 
-  a = m_outer_radius;
-  b = m_inner_radius;
-  c = m_layer_width;
-  a = a * c;
-  b = b * c;
+  auto const c = m_layer_width;
+  auto const a = m_outer_radius * c;
+  auto const b = m_inner_radius * c;
 
   /***** Convert 3D coordinates to 2D planar coordinates *****/
 
   // Calculate the point on position + mu * orientation,
   // where the difference segment is orthogonal
 
-  mu = (m_orientation * pos - m_position * m_orientation) /
-       m_orientation.norm2();
+  auto const mu = (m_orientation * pos - m_position * m_orientation) /
+                  m_orientation.norm2();
 
   // Then the closest point to the line is
 
-  closest_pos = m_position + mu * m_orientation;
+  auto const closest_pos = m_position + mu * m_orientation;
 
   // So the shortest distance to the line is
 
-  Utils::Vector2d dist_2D;
-  dist_2D[0] = (closest_pos - pos).norm();
-  dist_2D[1] = mu * m_orientation.norm();
+  auto const dist_2D =
+      Utils::Vector2d{{(closest_pos - pos).norm(), mu * m_orientation.norm()}};
```

Comment:
Is there a specific reason why you are using two pairs of braces for initialization? `Vector2d{0., 1.}` is totally equivalent, with the difference that `Array<double, 2>{0., 1.}` would work two (using aggregate init). See also [the discussion in the core guidlines](https://github.com/isocpp/CppCoreGuidelines/blob/master/CppCoreGuidelines.md#es23-prefer-the--initializer-syntax) on initializer synatax.

Reply:
More of a habit rather than a specific reason. This avoids ambiguous code in situations where pointers are involved, because a ctor call with two pointers can be interpreted as begin/end iterators:
```c++
#include<vector>
int main() {
    int *a = nullptr, *b = nullptr;
    std::vector<int> vec{a, b};  // segfault, no compiler warning
    std::vector<int> vec{{a, b}};  // compiler warning: type mismatch
}
```
This is also more coherent with what you would write for e.g.
```c++
struct A {int a[2]; int b;};
struct A object{{1,2}, 3};
```
i.e. the outer curly braces call the ctor, the inner curly braces delimit the first argument to the ctor.

I took this habit without really looking into style guides. It also solves an issue that is rather unlikely to occur. Should I refrain from using this syntax from now on?

Reply:
Well you made me ask myself and then you why there are additional braces. Also the single braces are easier teachable, because they work on all types on classes. I don't think the difference between a constructor, aggregate initialization and an initializer list is sth I'd teach to beginners. Instead I'd teach the uniform syntax that works almost always and does the right thing except in rare occasions. (But there is much debate on this area, see e.g. https://www.youtube.com/watch?v=7DTlWPgX6zs, also I think the rules changed again slightly with C++17...).

Reply:
I actually took this habit after watching two different CppCon short talks where the iterator issue forced speakers to use double braces in their codebase. I haven't watched the talk you just posted yet. Should I revert the double braces initialization?

Reply:
actually I can drop the double braces, they don't always help with pointers:
```c++
#include<vector>
#include<string>
int main () {
    std::vector<std::string> vec{{"1", "2"}};     // segfault: iterators
    std::vector<std::string> vec{"1", "2"};       // ok
    std::vector<std::string> vec{{"1"}, {"2"}};   // ok, "correct" way?
    std::vector<std::string> vec{{{"1"}, {"2"}}}; // ok
    return vec.size();
}
```